src/channel.cr

```diff
@@ -352,19 +405,18 @@ class Channel(T)
 
     ops_locks.each &.lock
 
-    # Check that no channel is closed
-    unless ops.all?(&.available?)
```

Comment:
If this is removed, then the `available?` is no longer used.

But I don't follow where the check that non of the channel is originally closed is done. Because if there is a channel ready and other closed I think the current code in this PR will yield and not raise.

Reply:
Because `op.execute` will return proper state, I think there is no need to check channel twice. The case you mentioned is a new case, and current spec doesn't cover it. The following code:

```crystal
# ch1 is ready and ch2 is closed

# a.cr
select
when m1 = ch1.receive?
when m2 = ch2.receive
end

# b.cr
select
when m1 = ch1.receive
when m2 = ch2.receive?
end
```

What are the expect behaviors of the two code pieces?

Reply:
Ok. We can avoid the double check and do not have a guarantee regarding which channel, the closed or the ready one, will fulfill the select.

* `a.cr` will either execute  ch1's `when` with `m1 = nil`, or ch2's `when` with data.
* `b.cr` will either raise due to ch1 been closed, or ch2's `when` with data.

There are no specs mixing `receive` and `receive?` though. 

If `ch2` would not be ready I would expect that `b.cr` will **always** raise. Both on a blocking and non-blocking `select`. That is a spec that could be added. But I don't expect that `receive` and `receive?` are mixed.

So, this change is good.

Reply:
I see. Although your precondition is different from me, I will add the spec about mixing `receive` and `receive?`.